lib/vanagon/component/source/git.rb

```diff
@@ -45,7 +45,7 @@ def initialize(url, workdir:, **options)
           # Ensure that #url returns a URI object
           @url = URI.parse(url.to_s)
           @ref = opts[:ref]
-          @workdir = workdir
+          @workdir = File.realpath(workdir) if workdir
```

Comment:
I don't think so. Everywhere this code is called requires a workdir to be passed, I was mostly running into issues with unit tests where workdir was set to `nil`. I should probably just update the tests to have an actual workdir rather than just setting workdir to nil.